### ISSUE TITLE
GCC 12 warns about MetaModelica's use of `strcmp`

### ISSUE THREAD

**Issue by perost** [OpenModelica/OpenModelica#9035]
GCC has started to complain about how MetaModelica uses `strcmp` when matching strings, producing a ton of messages like this:
```
OM/build/OMCompiler/Compiler/c_files/NFExpression.c:1872:51: warning: 'strcmp' reading 1 or more bytes from a region of size 0 [-Wstringop-overread]
 1872 |                 case 1: {
      |                                                   ^                                                        
.OM/OMCompiler/SimulationRuntime/c/meta/meta_modelica_data.h:185:18: note: at offset 8 into source object 'header' of size 8
  185 |       mmc_uint_t header;    \
      |                  ^~~~~~
OM/build/OMCompiler/Compiler/c_files/NFExpression.c:16:14: note: in expansion of macro 'MMC_DEFSTRINGLIT'
   16 | static const MMC_DEFSTRINGLIT(_OMC_LIT_STRUCT3,11,_OMC_LIT3_data);
      |              ^~~~~~~~~~~~~~~~
```
I guess the issue might be that `MMC_DEFSTRINGLIST` defines a new struct with a data member of the appropriate size, which `MMC_STRINGDATA` then cast to a `mmc_string` that's defined as:
```
struct mmc_string {
    mmc_uint_t header;  /* MMC_STRINGHDR(bytes) */
    char data[1];  /* `bytes' elements + terminating '\0' */
};
```
I'm not sure how that even works but apparently it does. I'm also not sure how to fix that, but if we're sure this is fine then we should at least add the `-Wstringop-overread` flag to the compilation flags.

**Comment by mahge** [comment]
It's been a while and I do not think this is going to be fixed. Should we just go ahead and silence the warning?

**Comment by sjoelund** [comment]
Or maybe it should just be char *data?

**Comment by sjoelund** [comment]
Or wait... That breaks the struct, right? Maybe it just shouldn't cast to struct...

**Comment by perost** [comment]
Yeah, I tried changing it to `char *data` and then the bootstrapped compiler just segfaults.

**Comment by sjoelund** [comment]
Because it's supposed to be a data segment that is allocated within the data structure. There is no indirection... The struct is probably the wrong thing to cast to.

**Comment by perost** [comment]
I had another look at this since the huge amount of warnings we get makes it hard to see actually legitimate warnings. The code should be fine but I couldn't find any way to get GCC to understand that, and due to all the macros there's no good way to selectively disable the warnings.

So #11983 disables the `stringop-overread` flag for the CMake build. If anyone wants to add it to the autotools build feel free to do so, I don't know where or how to add compiler-specific flags to it and don't use it anyway.
